### ISSUE TITLE
Feature Request: progress bar

### ISSUE THREAD

**Issue by kevinlinxc** [MeViMo/youbit#17]
It would be nice if there was a progress bar, e.g. using https://github.com/tqdm/tqdm, while the video is encoding. The encoding can take some time.

**Comment by kevinlinxc** [comment]
I can test something simple out and make a PR!

**Comment by flonle** [comment]
Yes, a progressbar would definitely be a nice addition. I'd approve a PR for it, but it won't be easy, there's no single loop you can use to report progress back to tqdm. Using multiple progress bars for different parts of the encoding/decoding progress might be better than trying to weigh the different parts against each other in one single progressbar.
For the uploading step you can scan the YouTube webpage for upload progress. Downloading is with an API so progress reports should be relatively easy to acquire.

**Comment by kevinlinxc** [comment]
Yeah, I realized because of how the chunking generator works its tricky.
